### ISSUE TITLE
Show all tasks on profile

### ISSUE THREAD

**Issue by diannakane** [medic/medic-webapp#1852]
For the Tasks tab, can we retain the 7 days max for displayed tasks, but in the Contacts tab, when you select a Family, you should see ALL tasks associated with that family regardless of the followup duration

MM comments: I think this is an interesting and useful request, but there are some design issues to be figured out. Eg do you show all the tasks, or just the next one outside of the 7 day window. What happens if you complete a task outside of the window? Should it not let you complete it? Or should it be completed as normal meaning tasks can be done at any time - which defeats the purpose of the time window.

For the field test we can either show just the ones in the time window, or all - but wont be able to clear them and without any info on when they can... because we removed that page as per earlier feedback.

Waiting for UX decisions - what is the issue they're experiencing?  We're actually showing tasks that _can be done now_

**Comment by diannakane** [comment]
Work-in-progress mockups: 

https://moqups.com/#!/edit/joshnesbit/1fF4IwJA
https://moqups.com/#!/edit/joshnesbit/1fF4IwJA

**Comment by Lesterng** [comment]
Needs work sessions and discussions

**Comment by Lesterng** [comment]
**Household tasks for enire housrhold / family**
https://invis.io/XA7CTCBGU#/163504245_Area-household
<img width="1075" alt="screen shot 2016-06-08 at 4 30 28 pm" src="https://cloud.githubusercontent.com/assets/5149601/15914579/f52d3966-2d96-11e6-910d-f2533dd275ac.png">


**Person tasks for individuals**
https://invis.io/XA7CTCBGU#/163504603_Area-person
<img width="1306" alt="screen shot 2016-06-08 at 4 27 38 pm" src="https://cloud.githubusercontent.com/assets/5149601/15914560/cea0fd1e-2d96-11e6-8792-a9acf19f9a9c.png">

**Comment by diannakane** [comment]
Description of how tasks on profiles work:

For this iteration, Tasks will only appear on type: clinic place profile for Android projects (i.e. households) and people belonging to that place (i.e. family members). 

-- Tasks to be completed for this place (household) and people (family members) belonging to that place appear in a consolidated list on the place's profile.
-- On an individual person's profile, only tasks be completed for that person (family member) will appear. 
-- Task card includes:
{{ Name }} The person or place to which the task refers, e.g. "Alison Jones" or "Alison Jones Family"
{{ Title }} Of the task, e.g, "ANC visit follow up"
{{ Task Status }} options include: "Past Due" with red dot, or simply a green dot
[ {{ # days overdue }} days ago ] OR [ in {{ # days until due }} days ]
-- Task filters appear above the list of tasks:
1 week  |  2 weeks  |  View all (separated by a vertical line)
Filters default to "1 week" view
-- There is no limit the number of tasks that can appear in this section. 
-- Clicking on a task opens that form associated with that task within the RHS. ** this may transition to a modal. 

@Lesterng let me know if I missed anything.

**Comment by abbyad** [comment]
@amandacilek do we want to revisit this?

**Comment by amandacilek** [comment]
Hm, I think Dianna's comment from June 9th is already implemented (tasks for family members display on the family profile). Her first comment about the time window of tasks displayed I think is still relevant. This relates to other Tasks changes we have been thinking about (like potentially getting rid of the due date altogether). 

I think I'd like to wait to discuss this one with Dianna again when she gets back before making any decisions.

**Comment by amandacilek** [comment]
@abbyad we think this true but we'd like to clarify: Tasks on the Tasks tab are restricted to a certain time frame (e.g. it doesn't show every single task scheduled months and years out), while tasks on the profile will be shown no matter how far in the future they are scheduled by clicking on "View All".

**Comment by abbyad** [comment]
Tasks have a time window, which has a start date and end date. They also have a due date which must be in between. Wherever you are in the app you can only see tasks when the current time is between the start and end dates (inclusive). On a person's profile, the "1 week" and "2 weeks" further limits that to tasks that have a _due date_ within 1 week, or 2 weeks. With "View all" you again see all tasks for which you are in the time window, regardless of the due date. I don't think there is anywhere in the app to see tasks before you enter their time window.

@garethbowen can you confirm that is your understanding too, and that this is how it is implemented?

**Comment by garethbowen** [comment]
@abbyad Yes although it's useful to note that the task window is only used in nools, that is to say, any given project may not have a window defined either intentionally or unintentionally. If the window is not defined, then this "view all" button could potentially show a lot of tasks. That would probably indicate an error in configuration and could be solved in configuration without requiring a limit in the webapp code.
